hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -338,6 +345,35 @@ protected Flow executeFromState(MasterProcedureEnv env, RegionStateTransitionSta
     try {
       switch (state) {
         case REGION_STATE_TRANSITION_GET_ASSIGN_CANDIDATE:
+
+          // Need to do some sanity check for replica region, if the region does not exist at file
+          // system, do not try to assign the replica region, log error and return.
+          // Do not rely on master's in-memory state, primary region got its own life, it can be
+          // closed, offline for various reasons.
```

Comment:
This is a mistake then? Master is the authority. If it can't make a call on read replicas, then read replicas are not done right?

Reply:
Probably you are right here. The defense in master can be changed to check its in-memory data structures, if master is not aware of the primary region, then there is something wrong. I will move this check to region server' region open.

Reply:
Updated to check primary region's inmemory state for defense.